src/PrestaShopBundle/Routing/Exception/RouteNotFoundException.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Routing\Exception;
+
+use PrestaShop\PrestaShop\Core\Exception\ErrorMessageException;
+use Throwable;
+
+/**
+ * Class RouteNotFoundException.
+ */
+class RouteNotFoundException extends ErrorMessageException
+{
+    /**
+     * RouteNotFoundException constructor.
+     *
+     * @param string $key
+     * @param array $parameters
+     * @param null $domain
+     * @param int $code
+     * @param Throwable|null $previous
+     */
+    public function __construct($key, array $parameters = [], $domain = null, $code = 0, Throwable $previous = null)
+    {
+        parent::__construct($key, $domain, $parameters, $code, $previous);
```

Comment:
Why do you override the `__construct` if you copy the parent behavior ?

Reply:
it change $parameters and $domain order, but don't know why :/

Reply:
be cause I didn't have a domain for this exception, but it is a bad idea to leave it empty, and this kind of exception is not displayed as an error message or notification
So I changed it to a more common CoreException